shared_model/builders/protobuf/builder_templates/query_template.hpp

```diff
@@ -123,6 +133,22 @@ namespace shared_model {
         if (first_hash) {
           page_meta_payload->set_first_tx_hash(first_hash->hex());
         }
+        if (first_tx_time) {
+          auto timestamp_begin = page_meta_payload->mutable_first_tx_time();
+          timestamp_begin->set_seconds(first_tx_time.value() / 1000);
```

Comment:
the code is duplicated twice, I suggest to extract method/function to setup timestamp from tx_time. I also suggest to change the magic numbers 1000 and 1000000 into `milliseconds_per_second`, similar in another case